lib/checks/aria/required-parent.js

```diff
@@ -62,26 +79,26 @@ function getAriaOwners(element) {
 	return owners.length ? owners : null;
 }
 
-var missingParents = getMissingContext(virtualNode);
-
-if (!missingParents) {
-	return true;
-}
+/**
+ * Recursively attempt to find parent with a desired role
+ * @param {HTMLElement} elm element
+ * @param {String} role aria role
+ * @returns {Boolean}
+ */
+function hasParentWithRole(elm, role) {
+	const parent = dom.getComposedParent(elm);
+	if (!parent) {
+		return false;
+	}
 
-var owners = getAriaOwners(node);
+	const parentRole = aria.getRole(parent, { noImplicit: true });
```

Comment:
Implicit roles should probably be considered. Current axe does not fail this:

```
<ul>
  <div role="listitem">List item 1</div>
</ul>
```

but these changes will.